src/librustdoc/doctest.rs

```diff
@@ -803,6 +805,22 @@ fn run_test(
     let duration = instant.elapsed();
     if doctest.no_run {
         return (duration, Ok(()));
+    } else if doctest.langstr.should_panic
+        // Equivalent of:
+        //
+        // ```
+        // (cfg!(target_family = "wasm") || cfg!(target_os = "zkvm"))
+        //     && !cfg!(target_os = "emscripten")
+        // ```
+        && let TargetTuple::TargetTuple(ref s) = rustdoc_options.target
```

Comment:
We both know that this check is super brittle and icky :D We can ignore that for now I guess ^^'

Okay, so I'm an utter noob when it comes to target specifications but ideally this would also handle `TargetJson`, right? The user may pass `--target=custom.json` which may specify the `panic_strategy` (IIUC if that's set to `Some(Abort | ImmediateAbort)` then that implies that unwinding panics are not supported?) as well as `target_family` and so on.

If what I write makes sense, could you at least leave a FIXME that we should somehow support `TargetJson` here, too?

Ideally, we would use some preexisting compiler API for querying this (one that provides a richer representation of the target), maybe there is one we could use?